File: codegen/results/v001/qwen3.5_4b_nothink/20260408_194201/go/008_prime_numbers/solution.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func main() {
	if len(os.Args) != 2 {
		fmt.Fprintf(os.Stderr, "Usage: %s <integer>\n", os.Args[0])
		os.Exit(1)
	}

	nStr := os.Args[1]
	n, err := strconv.Atoi(nStr)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error: expected a valid integer, got '%s'\n", nStr)
		os.Exit(1)
	}

	if n < 2 {
		// No primes less than 2
		return
	}

	// Sieve of Eratosthenes approach to find primes up to n
	isPrime := make([]bool, n+1)
	for i := 2; i <= n; i++ {
		isPrime[i] = true
	}

	for i := 2; i*i <= n; i++ {
		if isPrime[i] {
			for j := i * i; j <= n; j += i {
				isPrime[j] = false
			}
		}
	}

	for i := 2; i <= n; i++ {
		if isPrime[i] {
			fmt.Println(i)
		}
	}
}

File: codegen/results/v001/qwen3.5_4b_nothink/20260408_194201/python/008_prime_numbers/solution.py
import sys

def is_prime(n):
    """Check if a number is prime."""
    if n < 2:
        return False
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    for i in range(3, int(n**0.5) + 1, 2):
        if n % i == 0:
            return False
    return True

def main():
    # Get N from command line arguments
    if len(sys.argv) != 2:
        print("Usage: python script.py N", file=sys.stderr)
        sys.exit(1)
    
    try:
        n = int(sys.argv[1])
    except ValueError:
        print(f"Error: Expected an integer, got '{sys.argv[1]}'", file=sys.stderr)
        sys.exit(1)
    
    # Ensure N is non-negative
    if n < 0:
        return

    # Generate and print primes up to N
    primes = []
    for num in range(2, n + 1):
        if is_prime(num):
            primes.append(num)
    
    for prime in primes:
        print(prime)

if __name__ == "__main__":
    main()